content/chat/participant-activity.textile

```diff
@@ -1,7 +1,220 @@
+
+```[javascript]
+channel.presence.subscribe(‘leave’, (member) => {
+  user.online = false
+  user.lastOnline = member.timestamp
+
+  // Custom functionality to update display for chat participant that another participant is no longer online
+});
+```
+
+h3(#update). Update
+
+In certain chat instances, chat members can associate their chat participant status with a text-based status. For example, if you have a chat participant object (user) with the @statusMessage@ field, you can trigger an @update@ event using the code sample below to update this user with a new text-based string.
+
+```[javascript]
+channel.presence.update('<NEW_STATUS_UPDATE>', (err) => {
+  user.statusMessage = '<NEW_STATUS_UPDATE>';
+});
+```
+
+When a chat participant triggers the @update@ event all subscribers for this presence channel will be notified in real-time allowing them to update the string value of that chat participant's status of this user for everyone.
+
+```[javascript]
+channel.presence.subscribe('update', (memberUpdate) => {
+  user.status = memberUpdate.data
+});
+```
+
+h3(#other-events). Other events / more information
+
+The @present@ event is not relevant for chat participant statuses, however, if you wish to read more about other functionalities of presence, please refer to the "presence":/presence-occupancy/presence?lang=javascript#trigger-events docs page.
+
+h2(#subscribe). Subscribe chat participants to events
+
+Chat participant clients can subscribe to the presence set, enabling them to receive updates on other participants and update the UI accordingly (e.g., displaying online status icons).
+
+The message count within a presence set, like online status, can surge as the participant number grows. For instance, with 200 participants rapidly joining and leaving, the potential messages for online status alone could reach 80,400. The breakdown includes:
+
+200 chat participant messages published for the @enter@ event.
+200 × 200 (40,000) messages subscribed to for the @enter@ events.
+200 chat participant messages published for the @leave@ event.
+200 × 200 (40,000) presence messages subscribed to for the @leave@ event.
+
+For improved scalability, consider making your backend server the exclusive subscriber to chat participants' online status updates if your chat application anticipates a large number of participants. In this scenario, frontend clients can periodically request updates from the backend.
+
+h3(#batch-presence). Batch presence retrieval
+
+It is possible to reduce realtime presence updates received and traffic between clients by introducing batches using the @request()@ in the REST API. Within a batch response is an array of presence states, where each presence state update contains information about the clients in the presence set, including their @clientId@, member data, and "presence action":/docs/api/realtime-sdk/types#presence-action .
```

Comment:
This doesn't necessarily reduce the number of updates, unless you don't have all members subscribed to the set themselves. I do wonder whether we should include batch or not, as it's more of a use-case for returning the presence set from multiple channels. 

It would be a case of this vs occupancy to display the approximate membership of all 'chat rooms', and I expect occupancy may be the more efficient route. 

I think we should think about this one and decide whether to include it in chat.

Reply:
But, by batching the presence set, wouldn't this allow you to set intervals of when to retrieve all of the updates to the presence set and handle them at those points, rather than constantly receive individual updates as and when they happen? Reducing the number of traffic received to each client overall? Or rather reducing the quantity of traffic.